docs/features.md

```diff
@@ -874,30 +874,100 @@ For the `AuthConfig` above,
-- a fixed comparable `value`
+An expression contains one or more patterns and they must either all evaluate to true ("AND" operator, declared by grouping the patterns within an `all` block) or at least one of the patterns must be true ("OR" operator, when grouped within an `any` block.) Patterns not explicitly grouped are AND'ed by default.
+
+To avoid repetitions when listing patterns, any set of literal `{ pattern, operator, value }` tuples can be stored at the top-level of the AuthConfig spec, indexed by name, and later referred within an expression by including a `patternRef` in the block of conditions.
+
+**Examples of `when` conditions**
+
+i) to skip an entire `AuthConfig` based on the context (AND operator assumed by default):
+
+```yaml
+spec:
+  when: # auth enforced only on requests to POST /resources/*
+  - selector: context.request.http.method
+    operator: eq
+    value: POST
+  - selector: context.request.http.path
+    operator: matches
+    value: ^/resources/.*
+```
+
+ii) equivalent to the above with explicit AND operator (i.e., `all` block):
+
+```yaml
+spec:
+  when: # auth enforced only on requests to POST /resources/*
+  - all:
+    - selector: context.request.http.method
+      operator: eq
+      value: POST
+    - selector: context.request.http.path
+      operator: matches
+      value: ^/resources/.*
+```
+
+iii) OR condition (i.e., `any` block):
+
+```yaml
+spec:
+  when: # auth enforced only on requests with HTTP method equals to POST or PUT
+  - any:
+    - selector: context.request.http.method
+      operator: eq
+      value: POST
+    - selector: context.request.http.method
+      operator: eq
+      value: PUT
+```
 
-Literal expressions and references to expression sets (`patterns`, defined at the upper level of the `AuthConfig` spec) can be listed, mixed and combined in `when` conditions sets.
+iv) complex expression with nested operations:
 
-_Conditions_ can be used, e.g.,:
+```yaml
+spec:
+  when: # auth enforced only on requests to POST /resources/* or PUT /resources/*
+  - any:
+    - all:
+      - selector: context.request.http.method
+        operator: eq
+        value: POST
+      - selector: context.request.http.path
+        operator: matches
+        value: ^/resources/.*
+    - all:
+      - selector: context.request.http.method
+        operator: eq
+        value: PUT
+      - selector: context.request.http.path
+        operator: matches
+        value: ^/resources/.*
+```
 
-i) to skip an entire `AuthConfig` based on the context:
+v) more concise equivalent of the above (with implicit AND operator at the top level):
 
 ```yaml
 spec:
-  when: # no authn/authz required on requests to /status
+  when: # auth enforced only on requests to /resources/* path with method equals to POST or PUT
   - selector: context.request.http.path
-    operator: neq
-    value: /status
+    operator: matches
+    value: ^/resources/.*
+  - any:
+    - selector: context.request.http.method
```

Comment:
nit: is this the best example? Seems it could be achieved also with a `matches` operator? without too complex of a regex?

Reply:
Indeed, many cases of disjunction can be solved with the `matches` operator, and this is definitely true for patterns limited to finite sets (and many other infinite ones as well!), such as the values of the HTTP method. For other cases, users may want/have to go with `any`.

So I wanted to present `any` also as an alternative to `matches`.

<details>
<summary>Click here if you want to learn more about the limitations of the <code>matches</code> operator.</summary>

Because Golang [`regexp`](https://pkg.go.dev/regexp) package (behind our `matches` operator) is a strict implementation of regular expressions, in the theoretical computer science sense of the term (i.e., the kind that defines a [regular language](https://en.wikipedia.org/wiki/Regular_language)), some patterns often supported by other programming languages are simply not available in Go by default.

For details, see https://github.com/google/re2/wiki/Syntax.
</details>

But that was not the only reason though. There are yet other cases where the disjunction involves patterns with different selectors. E.g.: _If (method == &lt;X&gt;) OR (path =~ &lt;Y&gt;.*)_.

I could have used any one of those more complex cases here. I was trying to reduce the cognitive complexity of the example by going with something simple and straightforward. I guess I didn't account for the more skilled readers, like you, who often knows a little bit more about the suject and therefore also other ways to solve the problem 🙂